keras_nlp/metrics/bleu.py

```diff
@@ -92,7 +92,86 @@ class Bleu(keras.metrics.Metric):
         dtype: string or tf.dtypes.Dtype. Precision of metric computation. If
                not specified, it defaults to tf.float32.
         name: string. Name of the metric instance.
-        **kwargs: Other keyword arguments.
+
+    Examples:
+
+    1. Various Input Types.
+    1.1. Python string.
+    >>> bleu = keras_nlp.metrics.Bleu(max_order=4)
+    >>> ref_sentence = "the quick brown fox jumps over the lazy dog"
+    >>> pred_sentence = "the quick brown fox jumps over the box"
+    >>> score = bleu([ref_sentence], [pred_sentence])
+    <tf.Tensor(0.7420885, shape=(), dtype=float32)>
+
+    1.2. rank 1 inputs.
+    a. Python list.
+    >>> bleu = keras_nlp.metrics.Bleu(max_order=4)
+    >>> ref_sentence = [
+        "the quick brown fox jumps over the lazy dog",
+        "the quick brown fox jumps over the lazy frog"
+    ]
+    >>> pred_sentence = ["the quick brown fox jumps over the box"]
+    >>> score = bleu(ref_sentence, pred_sentence)
+    <tf.Tensor(0.7420885, shape=(), dtype=float32)>
+
+    b. Tensor.
+    >>> bleu = keras_nlp.metrics.Bleu(max_order=4)
+    >>> ref_sentence = tf.constant([
+        "the quick brown fox jumps over the lazy dog",
+        "the quick brown fox jumps over the lazy frog"
+    ])
+    >>> pred_sentence = tf.constant(["the quick brown fox jumps over the box"])
+    >>> score = bleu(ref_sentence, pred_sentence)
+    <tf.Tensor(0.7420885, shape=(), dtype=float32)>
+
+    c. RaggedTensor.
+    >>> bleu = keras_nlp.metrics.Bleu(max_order=4)
+    >>> ref_sentence = tf.ragged.constant([
+        [
+            "the quick brown fox jumps over the lazy dog",
+            "the quick brown fox jumps over the lazy frog"
+        ]
+    ])
+    >>> pred_sentence = tf.ragged.constant([
+        ["the quick brown fox jumps over the box"]
+    ])
+    >>> score = bleu(ref_sentence, pred_sentence)
+    <tf.Tensor(0.7420885, shape=(), dtype=float32)>
```

Comment:
Since we mention that we are using ragged tensors here, it would be great if we could actually have a tensor with variable dimension along an axis. Also, I think your example will throw an error; `pred_sentence` should be a dense tensor here, not a ragged tensor.

So, something like:

```
>>> bleu = keras_nlp.metrics.Bleu(max_order=4)
>>> ref_sentence = tf.ragged.constant([
...         ["the quick brown fox jumps over the lazy dog"],
...         ["the quick brown fox jumps over the lazy dog", "the quick brown fox jumps over the lazy frog"]
...     ])
>>> pred_sentence = tf.constant([
...         ["the quick brown fox jumps over the box"],
...         ["the quick brown fox jumps over the box"]
...     ])
>>> bleu(ref_sentence, pred_sentence)
<tf.Tensor: shape=(), dtype=float32, numpy=0.7420885>
```